java-api-core/src/com/kinvey/java/store/BaseDataStore.java

```diff
@@ -390,39 +367,41 @@ public void pushBlocking() throws IOException {
         Preconditions.checkNotNull(client, "client must not be null.");
         Preconditions.checkArgument(client.isInitialize(), "client must be initialized.");
         Preconditions.checkArgument(client.getSyncManager().getCount(getCollectionName()) == 0, "InvalidOperation. You must push all pending sync items before new data is pulled. Call push() on the data store instance to push pending items, or purge() to remove them.");
-
-        KinveyAbstractReadResponse<T> response = new KinveyAbstractReadResponse<T>();
         query = query == null ? client.query() : query;
+        KinveyAbstractReadResponse<T> response = networkManager.pullBlocking(query, cache, isDeltaSetCachingEnabled()).execute();
+        cache.delete(query);
+        cache.save(response.getResult());
+        return response;
+    }
 
-        if (isAutoPaginationEnabled()) {
-            if (query.getSortString() == null || query.getSortString().isEmpty()) {
-                query.addSort(Constants._ID, AbstractQuery.SortOrder.ASC);
-            }
-            List<T> networkData = new ArrayList<T>();
-            List<Exception> exceptions = new ArrayList<Exception>();
-            int skipCount = 0;
-            int pageSize = this.pageSize;
-
-            // First, get the count of all the items to pull
-            int totalItemCount = this.countNetwork();
-            KinveyAbstractReadResponse<T> pullResponse;
-            do {
-                query.setSkip(skipCount).setLimit(pageSize);
-                pullResponse = networkManager.pullBlocking(query, cache, isDeltaSetCachingEnabled()).execute();
-                networkData.addAll(pullResponse.getResult());
-                exceptions.addAll(pullResponse.getListOfExceptions());
-                cache.delete(query);
-                cache.save(networkData);
-                skipCount += pageSize;
-            } while (skipCount < totalItemCount);
-            response.setResult(networkData);
-            response.setListOfExceptions(exceptions);
-        } else {
-            response = networkManager.pullBlocking(query, cache, isDeltaSetCachingEnabled()).execute();
-            cache.delete(query);
-            cache.save(response.getResult());
+    public KinveyAbstractReadResponse<T> pullBlocking(Query query, int pageSize) throws IOException {
+        Preconditions.checkArgument(pageSize > 0, "pageSize must be more than 0");
```

Comment:
Do we also need  a precondition here to verify that `pageSize <= 10000`? Also, is there a default page size set?

Reply:
In this implementation, we don't have default page size value because the SDK doesn't have `isAutoPagination` flag inside.